packages/typescript/src/utils/plugin-banner.ts

```
@@ -0,0 +1,354 @@
+/**
+ * Shared plugin banner utilities.
+ *
+ * WHY: Multiple plugins want custom ANSI banners, but the hard part is not the
+ * ASCII art itself. The tricky bits are terminal display width, full-width
+ * Unicode, emoji/grapheme handling, and ANSI-safe truncation/padding. Centralize
+ * those rules so plugins can focus on content and keep layout correct.
+ */
+
+import type { IAgentRuntime } from "../types/runtime";
+
+const ANSI_PATTERN = /\x1b\[[0-9;]*m/g;
```

Biome (2.4.4)

[error] Unexpected control character in a regular expression.

(lint/suspicious/noControlCharactersInRegex)